wrangler-service/src/main/java/io/cdap/wrangler/service/directive/AbstractDirectiveHandler.java

```diff
@@ -97,6 +107,76 @@ public void destroy() {
     return new CommonDirectiveExecutor(contextProvider, composite).executeDirectives(namespace, directives, sample);
   }
 
+
+  /**
+   * Formats Structure and Array records in a "pseudojson" in which it collapses
+   * complex types by callint toString().  This is done so the complex types are shown
+   * in a similar way inside and outside the structure records.
+   */
+  private static class WranglerDisplaySerializer implements TypeAdapterFactory {
+    @Override
+    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+      // Let GSON Handle primitives and collections
+      if (ClassUtils.isPrimitiveOrWrapper(type.getRawType())
+          || (Iterable.class.isAssignableFrom(type.getRawType()))) {
+        return null;
+      }
+      if (Row.class.isAssignableFrom(type.getRawType())) {
+        return (TypeAdapter<T>) rowClassAdapter(gson);
+      }
+
+      return (TypeAdapter<T>) defaultClassAdapter(gson);
+    }
+
+    private TypeAdapter<Row> rowClassAdapter(Gson gson) {
+      return new TypeAdapter<Row>() {
+        @Override
+        public void write(JsonWriter out, Row value) throws IOException {
+          if (value == null) {
+            return;
+          }
+          final TypeAdapter<Object> elementAdapter = gson.getAdapter(Object.class);
+          out.beginObject();
+          for (Pair<String, Object> field : value.getFields()) {
+            out.name(field.getFirst());
+            elementAdapter.write(out, field.getSecond());
+          }
+          out.endObject();
+        }
+
+        @Override public
+        Row read(JsonReader in) throws IOException {
+          throw new UnsupportedOperationException("Reading Rows from Wrangler display format not implemented");
+        }
+      };
+    }
+
+    private TypeAdapter<Object> defaultClassAdapter(Gson gson) {
+      final TypeAdapter<Object> elementAdapter = gson.getAdapter(Object.class);
+      return new TypeAdapter<Object>() {
+        @Override
+        public void write(JsonWriter out, Object value) throws IOException {
+          try {
+            if ((value.getClass().getMethod("toString").getDeclaringClass() != Object.class)) {
+              out.value(value.toString());
+            } else {
+              out.value("Non-displayable object");
```

Comment:
Could you please confirm with product on what to say here. I'd prefer something shorter as those JSONs going to be displayed as 1 line in UI